packages/fetchai/skills/aries_alice/behaviours.py

```diff
@@ -104,6 +110,31 @@ def teardown(self) -> None:
         self._unregister_service()
         self._unregister_agent()
 
+    def _retry_failed_registration(self) -> None:
+        """
+        Retry a failed registration.
+
+        :return: None
+        """
+        if self.failed_registration_msg is not None:
+            self._nb_retries += 1
+            if self._nb_retries >= self._max_soef_registration_retries:
+                self.context.is_active = False
+                return
+
+            oef_search_dialogues = cast(
+                OefSearchDialogues, self.context.oef_search_dialogues
+            )
+            oef_search_msg, _ = oef_search_dialogues.create(
+                counterparty=self.failed_registration_msg.to,
+                performative=self.failed_registration_msg.performative,
+                service_description=self.failed_registration_msg.service_description,
+            )
+            self.context.outbox.put_message(message=oef_search_msg)
+            self.context.logger.info("retrying registration on SOEF.")
```

Comment:
first time it's not retrying, so maybe amend

Reply:
This is no the first time. The first time is in `_register_agent` called in `setup()`. This is in `_retry_failed_registration` called from `act()` and only run if there is a failed registration message `if self.failed_registration_msg is not None`

Reply:
Ok, I'd add a counter to the log message (i.e. print `self._nb_retries`)